src/nvim/terminal.c

```diff
@@ -619,6 +619,21 @@ static void buf_set_term_title(buf_T *buf, char *title)
                &err);
 }
 
+// Get the associated buffer for the terminal and return whether it's in a
+// valid state to be used
+static bool term_buf_valid(Terminal *term, buf_T **buf)
+  FUNC_ATTR_NONNULL_ALL
+{
+  *buf = handle_get_buffer(term->buf_handle);
+  bool valid = buf_valid(*buf);
+  if (!valid) {
+    // Destroyed by `close_buffer`. Do not do anything else.
```

Comment:
Is the comment still valid? Should `close_buffer()` do something like set `term->buf_handle = 0`?